src/main/cpp/Auto/SwerveAutoPath.cpp

```diff
@@ -128,10 +215,17 @@ void AutoPath::UpdateOdom(vec::Vector2D curPos, double curAng) {
   m_curAng = curAng;
 }
 
+double SwerveAutoPath::calcGenericCompletion(double start, double current, double end) {
+  double frac = abs(current_distance - start) / abs(end - start);
```

Comment:
don't use absolute value (you don't need it). Also do the math correctly: (end - current)/(end - start), don't return 1.0 - that